src/util/simplify_expr2.cpp

```diff
@@ -415,6 +415,258 @@ static type2tc common_arith_op2_type(expr2tc &e, expr2tc &f)
 
 expr2tc add2t::do_simplify() const
 {
+  // X + X --> X << 1
+  if(is_symbol2t(side_1) && is_symbol2t(side_2))
+  {
+    if(to_symbol2t(side_1) == to_symbol2t(side_2))
+      return shl2tc(type, side_1, from_integer(1, type));
+  }
+
+  // (A + 1) + ~B --> A - B
+  auto simplify_1 = [](const expr2tc e, const expr2tc f) -> expr2tc {
+    auto sidecheck = to_add2t(e).side_2;
+    auto sidecheck2 = to_add2t(e).side_1;
+    if(to_constant_int2t(sidecheck).value == 1)
```

Comment:
It is not clear to me why the `side_2` of the add-expression has to have `constant_int2tc` type here. Can you explain?

Reply:
on this case sidecheck has to be equals to 1 and this is the way that i found to check this.

Reply:
Are you saying that the `if`-condition should only be true if the `side_2` is a constant integer with value 1? If so, please use `is_constant_int2t(sidecheck) && to_constant_int2t(sidecheck).value == 1` instead, since it might be that `sidecheck` is a different type of expression, e.g. the lhs in `(a + *p) + ~b`.